eng/testing/tests.targets

```diff
@@ -26,11 +26,15 @@
 
   <!-- Archive test binaries. -->
   <Target Name="ArchiveTests"
-          Condition="'$(ArchiveTests)' == 'true' and '$(IgnoreForCI)' != 'true'"
+          Condition="'$(ArchiveTests)' == 'true' and '$(IgnoreForCI)' != 'true' and ('$(TargetOS)' != 'iOS' and '$(TargetOS)' != 'tvOS' and '$(TargetOS)' != 'Android')"
           AfterTargets="PrepareForRun"
           DependsOnTargets="GenerateRunScript">
     <Error Condition="'$(TestArchiveTestsDir)' == ''" Text="TestArchiveTestsDir property to archive the test folder must be set." />
 
+    <PropertyGroup Condition="'$(TargetOS)' == 'Browser'">
+      <OutDir>$(OutDir)/AppBundle</OutDir>
```

Comment:
I think we should use another property for this... Something like:

```xml
<PropertyGroup>
   <_dirToCompress>$(OutDir)</_dirToCompress>
   <_dirToCompress Condition="'$(TargetOS)' == 'Browser'">$(OutDir)/AppBundle</_dirToCompress>
</PropertyGroup>
```

and then change ZipDirectory to take _dirToCompress as the SourceDirectory.

Also, we can move: https://github.com/dotnet/runtime/blob/d2f9adc5ffd50ff5df09090f77ae8364656dc37d/eng/testing/tests.mobile.targets#L4 into tests.targets so that we don't have to duplicate the AppBundle dir. 

Changing the `OutDir` property is not ideal.

Reply:
Sure, I was just testing my theory that this was why the wrong thing was happening. It's a bad fix as-is

Reply:
Yeah.. just adding the comments as I see stuff so that we don't forget addressing on the final iteration... just make it work, no need to address these comments now 😄